packages/blade/src/components/Input/BaseInput/BaseInputVisuals.tsx

```diff
@@ -290,14 +347,23 @@ export const BaseInputVisuals = ({
         {TrailingIcon ? (
           <BaseBox
             display="flex"
+            justifyContent="center"
+            alignItems="center"
             {...getTrailingIconStyles({ hasTrailingIcon, hasTrailingButton })}
           >
-            <TrailingIcon
-              size={iconSize[size]}
-              color={
-                isDisabled ? 'interactive.icon.gray.disabled' : trailingIconColor[validationState]
-              }
-            />
+            <ValidationIconTooltip
+              hasBorder={hasBorder}
+              errorText={errorText}
+              successText={successText}
+              validationState={validationState}
+            >
+              <TrailingIcon
+                size={iconSize[size]}
+                color={
+                  isDisabled ? 'interactive.icon.gray.disabled' : trailingIconColor[validationState]
```

Comment:
shouldn't this be surface.icon token? It doesn't have hover and selected state no?